src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -2063,6 +2064,69 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
         }
       } );
     }
+
+    if ( isTable && conn && conn->capabilities2().testFlag( Qgis::DatabaseProviderConnectionCapability2::SetTableComment ) )
+    {
+      QAction *editComment = new QAction( tr( "Set Comment…" ), menu );
+
+      addToSubMenu( menu, editComment, tr( "Manage" ) );
+
+      const QString connectionUri = conn->uri();
+      const QString providerKey = conn->providerKey();
+
+      connect( editComment, &QAction::triggered, editComment, [connectionUri, providerKey, item, context] {
+        QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( providerKey ) };
+        if ( !md )
+          return;
+
+        std::unique_ptr<QgsAbstractDatabaseProviderConnection> conn2( qgis::down_cast<QgsAbstractDatabaseProviderConnection *>( md->createConnection( connectionUri, QVariantMap() ) ) );
+
+        QString errCause;
+
+        const QString schemaName = item->parent()->name();
+        const QString tableName = item->name();
+
+        QString fullName;
+        if ( schemaName.isEmpty() )
+          fullName = tableName;
+        else
+          fullName = QStringLiteral( "%1.%2" ).arg( schemaName, tableName );
+
+        if ( conn2 )
+        {
+          const QString comment = conn2->table( schemaName, tableName ).comment();
+
+          QgsCommentInputDialog *dlg = new QgsCommentInputDialog( tr( "Table Comment" ), comment );
```

Comment:
dlg is leaked. You could just use:

```suggestion
          QgsCommentInputDialog dlg( tr( "Table Comment" ), comment );
```